yarp/yarp_compiler.c

```diff
@@ -516,6 +516,58 @@ yp_compile_class_path(LINK_ANCHOR *const ret, rb_iseq_t *iseq, const yp_node_t *
     }
 }
 
+/**
+ * In order to properly compile multiple-assignment, some preprocessing needs to
+ * be performed in the case of call or constant path targets. This is when they
+ * are read, the "parent" of each of these nodes should only be read once (the
+ * receiver in the case of a call, the parent constant in the case of a constant
+ * path).
+ */
+static void
+yp_compile_multi_write_lhs(rb_iseq_t *iseq, int lineno, const yp_node_t *node, LINK_ANCHOR *const ret, yp_compile_context_t *compile_context, uint8_t *pushed, bool nested) {
+    NODE dummy_line_node = generate_dummy_line_node(lineno, lineno);
+
+    switch (YP_NODE_TYPE(node)) {
+        case YP_MULTI_TARGET_NODE: {
+            yp_multi_target_node_t *cast = (yp_multi_target_node_t *) node;
+            for (size_t index = 0; index < cast->targets.size; index++) {
+                yp_compile_multi_write_lhs(iseq, lineno, cast->targets.nodes[index], ret, compile_context, pushed, false);
+            }
+            break;
```

Comment:
Can these `break`s be `return`s?

Reply:
I haven't changed this, because I refactored the code to rely on `pushed` being a return value rather than a pointer manipulation - `break`ing out of the `case` and having a single `return` reads better to me than having a `return pushed` inside each `case` branch.